docs/snakefiles/rules.rst

```diff
@@ -330,6 +330,92 @@ It's also possible to get named input/output files in the following way:
+~~~~~~~~~~~~~~
+
+Certain components in input and output file paths tend to reoccur across many rules.
+Via so-called **pathvars**, Snakemake allows to define such components globally, make them configurable via the config file, and change them per module or even per rule.
+Apart from saving boilerplate code, pathvars can be used to make modules intended for reuse in multiple contexts more flexible.
+Pathvars can be used as generic placeholders for their actual values inside of input, output, log, and benchmark paths, using angle brackets, e.g. ``<results>``.
+They behave similarly to `Python string interpolation <https://docs.python.org/3/tutorial/inputoutput.html#formatted-string-literals>`__ but only allow predefined placeholders, with precedence and configuration handled by Snakemake.
+
+Pathvar usage
+"""""""""""""
+
+An example rule using pathvars is the following:
+
+.. code-block:: python
+
+    rule somerule:
+        input:
+            "<results>/something/{sample}.txt"
+        output:
+            "<results>/processed/{sample}.txt"
+        shell:
+            "somecommand {input} {output}"
+
+Pathvars are resolved when the rule is parsed (before wildcard resolution and DAG construction).
+The values of pathvars can thereby even contain wildcards themselves.
+
+Pathvar defaults
+""""""""""""""""
+
+By default, Snakemake offers the pathvars ``results``, ``resources``, ``logs``, ``benchmarks``.
+Each of them is set to its respective name (i.e. the output file ``"<results>/processed/{sample}.txt"`` will be interpreted as ``"results/processed/{sample}.txt"``).
+
+Pathvar definition
+""""""""""""""""""
+
+Beyond the defaults, it is possible to define additional pathvars or customize the default definitions.
+This can happen in multiple ways, with the following precedence (from highest to lowest):
+
+1. Per rule, via the ``pathvars`` keyword.
+2. Per :ref:`module <snakefiles-modules>`, via the ``pathvars`` argument to the ``module`` directive or via the ``pathvars`` key in a config dict explicitly passed to the module.
+3. Globally, via the ``pathvars`` key in the config file or passed to the ``--config`` command line arguments.
+4. Globally, via the ``pathvars`` keyword at the top level of the Snakefile.
+
+Apart from :ref:`per-module pathvars <snakefiles-modules-pathvars>`, the most common way is to define them globally via the ``pathvars`` keyword:
+
+.. code-block:: python
+
+    pathvars:
+        per="{sample}"
+
+Above, we define a pathvar ``per`` and set it to the value ``{sample}``, thus defining a wildcard.
+Such a pattern can be helpful if you write a workflow that shall be reused as a module in various different ways within other workflows (e.g. thereby processing different items, samples, or something else).
+
+In order to overwrite pathvars per rule, they can be specified via the ``pathvars`` keyword inside a rule:
+
+.. code-block:: python
+
+    rule somerule:
+        input:
+            "<results>/something/{sample}.txt"
```

Comment:
It may be clearer here if input is:
`"<results>/something/<per>.txt"` since the pathvar `per` was defined as `{sample}` above.

Same for output.